rats-apps/src/python/rats/apps/_container.py

```diff
@@ -64,13 +71,62 @@ def get_group(
 
         yield from self.get_namespaced_group(ProviderNamespaces.GROUPS, group_id)
 
-    @abstractmethod
     def get_namespaced_group(
         self,
         namespace: str,
         group_id: ServiceId[T_ServiceType],
     ) -> Iterator[T_ServiceType]:
         """Retrieve a service group by its id, within a given service namespace."""
+        tates = annotations.get_class_annotations(type(self))
+        containers = tates.with_namespace(ProviderNamespaces.CONTAINERS)
+        groups = tates.with_group(namespace, group_id)
+
+        for annotation in groups.annotations:
```

Comment:
@elonp can you help me fix the autoid annotation test failures? i refactored this annotated container logic and i think i made the changes we discussed with the caching being part of the container methods instead of the service decorators.